spydrnet_physical/util/tests/test_ConnectPointList.py

```diff
@@ -0,0 +1,219 @@
+
+	def test_push_connection_down(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.push_connection_down (pt)
+		self.assertEqual(pt.level,"down")
+
+	def test_pull_connection_up(self):
+		'''This test checks whether the level of the connection is going one down'''
+		pt = ConnectPoint(2, 2, 2, 3)
+		self.pts.pull_connection_up(pt)
+		self.assertEqual(pt.level,"up")
+
+	def test_make_top_connection(self):
+		'''This test checks whether the connection is made with the top level'''
+		pass
+		#pts = ConnectPointList(5,10)
+		#pt = ConnectPoint(2, 2, 2, 3)
+		#pts.make_top_connection()
+		#self.assertEqual(pt.level,"same")
+
+	def test_reset_level(self):
+		'''This test checks whether the connection reset to the same level'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		self.pts.reset_level(pt)
+		self.assertEqual(pt.level,"same")
+
+	def test_hold_cursor(self):
+		'''This test checks whether the cursor holds its position when a connection is made in any coordonate'''
+		self.pts.cursor = (2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.assertEqual(self.pts.cursor,(2,3))
+
+	def test_release_cursor(self):
+		'''This test checks whether the cursor also moves its position when a connection is made in any coordonate'''
+		self.pts.set_cursor(2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.pts.release_cursor()
+		self.pts.move_x(2).move_y(4)
+		self.assertEqual(self.pts.cursor,(4,7))
+
+	def test_flip_V(self):
+		'''This test checks whether the connection is correctly flipped vertically'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("V")
+		self.assertEqual(point.connection,(2,-2,2,-3))
+
+	def test_flip_H(self):
+		'''This test checks whether the connection is correctly flipped horizontally'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("H")
+		self.assertEqual(point.connection,(-2,2,-2,3))
+
+	def test_sample_connections(self):
+		'''This test checks whether the connection is sampled correctly equal to the max distance defined '''
+		pt = ConnectPoint(2, 2, 5, 2,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_to_point((5,2))
+		self.assertEqual(point.connection,(2,2,5,2))
+		pts.sample_connections(max_distance=1)
+		self.assertEqual(point.connection,(4,2,5,2))
+
+	def test_merge(self):
+		'''This test cheks whether two connect point lists are merged correclty'''
+		pts1 = ConnectPointList(5, 5)
+		pts1.cursor = (2,2)
+		pts1.move_x()
+		pts2 = ConnectPointList(5,5)
+		pts2.cursor = (2,3)
+		pts2.move_y()
+		pts1.merge(pts2)
+		point = pts1.search_from_point((2,3))
+		self.assertEqual(point.connection, (2, 3, 2, 4))
```

Comment:
use the method we discussed yesterday, convert connect point list to tuple and check f all the points from pts1 and pts2 exists in the merged version